lib/libyui_client/validate.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  module Validate
+    module_function
+
+    def filter(filter)
+      filters = %i[id label type].freeze
+      raise ArgumentError, "Invalid filter format: #{filter.inspect}. Hash is expected." unless filter.is_a? Hash
+
+      filter.delete_if { |_key, value| value.nil? }
```

Comment:
Don't we want to raise error in this case too? We also could perform validation in the single cycle below.

Reply:
I will remove it.